hook.py

```diff
@@ -19,72 +22,177 @@
+    repo_dir = 'plugins/atomic/atomic-red-team/'
+    defaults = {key: val for key, val in test.get("input_arguments", {}).items()}
     while RE_VARIABLE.search(string):
         full_string, varname = RE_VARIABLE.search(string).groups()
-        string = string.replace(full_string, defaults[varname])
+        if varname not in defaults:
+            # we did not find the default value of a variable
+            continue
+        default_var = str(defaults[varname]["default"])
+
+        # the variable is a path and refers to something in the atomics folder,
+        # possibly a payload
+        if "PathToAtomicsFolder" in default_var and defaults[varname]["type"].lower() == "path":
+            default_var = default_var.replace("PathToAtomicsFolder", "atomics")
+            if platform == "windows":
+                default_var = default_var.replace("\\", "/")
+            # TODO handle folders
+            if os.path.isfile(repo_dir+default_var):
+                full_path_attachement = repo_dir+default_var
+                default_var = _handle_attachments(full_path_attachement)
+                payloads.append(default_var)
+
+        string = string.replace(full_string, default_var)
 
-    return string
+    return string, payloads
 
 
 def _handle_multiline_commands(cmd):
     return cmd.replace("\n", ";")
 
 
-async def _prepare_executor(test):
-    command = _use_default_inputs(test, test['executor']['command'])
+async def _prepare_executor(entries, test, platform):
+    payloads = []
+
+    command, new_payloads = _use_default_inputs(entries, test, platform, test['executor']['command'])
+    payloads.extend(new_payloads)
     command = _handle_multiline_commands(command)
-    cleanup = _use_default_inputs(test, test['executor'].get('cleanup_command', ''))
-    cleanup = _handle_multiline_commands(cleanup)
 
-    encoded_command = BaseWorld.encode_string(command)
-    encoded_cleanup = BaseWorld.encode_string(cleanup)
+    cleanup, new_payloads = _use_default_inputs(entries, test, platform, test['executor'].get('cleanup_command', ''))
+    payloads.extend(new_payloads)
+    cleanup = _handle_multiline_commands(cleanup)
 
-    return (encoded_command, encoded_cleanup)
+    # TODO handle ART local files in command
+    # eg. https://github.com/redcanaryco/atomic-red-team/blob/a956d4640f9186a7bd36d16a63f6d39433af5f1d/atomics/T1022/T1022.yaml
+    # eg. https://github.com/redcanaryco/atomic-red-team/blob/fa67d8f04188955ec2c3d1532d2212f26ef5e3e8/atomics/T1502/T1502.yaml
+    return (command, cleanup, payloads)
 
 
-async def _save_ability(data_svc, entries, test):
+async def _save_ability(technique_to_tactics, entries, test):
     ability_id = hashlib.md5(json.dumps(test).encode()).hexdigest()
     at_ingested = 0
     at_total = 0
+
+    tactics_li = technique_to_tactics.get(entries['attack_technique'], ['redcanary-unknown'])
+    if len(tactics_li) > 1:
+        tactic = 'multiple'
+    else:
+        tactic = tactics_li[0]
+
+    data = {
```

Comment:
use the other dict definite format everywhere in your file
```Python
data = dict(id=ability_id, name=...)
```

Reply:
fixed in 6ea28af